cassdegrees/templates/list.html

```diff
@@ -32,35 +32,57 @@
-                                    (<a href="/api/model/{{ title.lower }}/{{ item }}/">View</a>)
-                                {% endif %}
-                            </td>
+                        {% for key in content.0 %}
+                            {% if key != "id" %}
+                                <th>{{ key }}</th>
+                            {% else %}
+                                <th>Select</th>
+                                <th>{{ title }}</th>
+                            {% endif %}
                         {% endfor %}
                     </tr>
-                {% endfor %}
-            </table>
-            {# If there is no data available, notify the user #}
-            {% if not content %}<p>No {{ title }}s to display</p>{% endif %}
+                    {# Add rows containing the contents of each json data instance #}
+                    {% for row in content %}
+                        <tr>
+                            {% for key, item in row.items %}
+
+                                    {# If the item is in the ID column, generate a link to it instead #}
+                                    {% if key != "id" %}
+                                        <td>{{ item }}</td>
+                                    {% else %}
+                                        <td><input title="Select" type="radio" name="id" value="{{ item }}"  /></td>
+                                        <td>(<a href="/api/model/{{ title.lower }}/{{ item }}/">View</a>)</td>
+                                    {% endif %}
+
+                            {% endfor %}
+                        </tr>
+                    {% endfor %}
+                </table>
+                {# If there is no data available, notify the user #}
+                {% if not content %}
+                    <p>No {{ title }}s to display</p>
+                {% else %}
+                    <div class="right">
+                        <p class="left marginright">
+                           <input class="btn-uni-grad btn-large" type ="submit" formaction="/{% if title == 'Degree' %}{% elif title == 'Subplan' %}{% elif title == 'Course' %}manage_courses{% endif %}/?action=Add" value="Add a {{ title }}">
+                        </p>
+                        <p class="left marginright">
+                           <input class="btn-uni-grad btn-large" type ="submit" formaction="/{% if title == 'Degree' %}{% elif title == 'Subplan' %}{% elif title == 'Course' %}manage_courses{% endif %}/?action=Edit" value="Edit Selected">
+                        </p>
+                        <p class="right">
+                           <input class="btn-uni-grad btn-large" type ="submit" formaction="/{% if title == 'Degree' %}{% elif title == 'Subplan' %}{% elif title == 'Course' %}manage_courses{% endif %}/?action=Delete" value="Delete Selected">
+                        </p>
+                   </div>
+                </form>
+            {% endif %}
```

Comment:
Wait, this endif ends *after* the form? That might break cases where there is no content.  (i.e <form> is unmatched).

Reply:
Fixed, didn't see that one.